### DISCUSSION TITLE
Negative rate - raw value does not match recognized value

### DISCUSSION THREAD

**Discussion by p0macs** [jomjol/AI-on-the-edge-device#3265] [General]
### The Problem

On my watermeter, I often have readings like this:

`2024-09-26T09:22:11+0200,main,569.5726,,579.6653,,0.0000,Neg. Rate - Read: - Raw: 569.5726 - Pre: 579.6653 ,4.9,7.4,9.7,6.2,7.3,2.5,6.0

2024-09-26T09:27:11+0200,main,579.5728,,579.6653,,0.0000,Neg. Rate - Read: - Raw: 579.5728 - Pre: 579.6653 ,4.9,7.4,9.6,6.1,7.3,2.7,8.1`

Can you explain why the raw value is 569.5726 - while the recognized values are: 
4.9 -> 5
7.4 -> 6 (????)
9.7 -> 9
6.2 -> 5 (????)
7.3 -> 7
2.5 -> 2
6.0 -> 6

And again, why the raw value is 579.5728 - while the recognized values are: 
4.9 -> 5
7.4 -> 7
9.6 -> 9
6.1 -> 5 (????)
7.3 -> 7
2.7 -> 2
8.1 -> 8

### Version

Release: v15.7.0 (Commit: 0d0b018+)

### Logfile

```shell
[0d20h46m28s] 2024-09-26T09:58:01 <ERR> [POSTPROC] main: Raw: 579.5728, Value: , Status: Neg. Rate - Read: - Raw: 579.5728 - Pre: 579.6653
[0d20h51m29s] 2024-09-26T10:03:02 <ERR> [POSTPROC] main: Raw: 579.5728, Value: , Status: Neg. Rate - Read: - Raw: 579.5728 - Pre: 579.6653
[0d20h56m28s] 2024-09-26T10:08:01 <ERR> [POSTPROC] main: Raw: 579.5728, Value: , Status: Neg. Rate - Read: - Raw: 579.5728 - Pre: 579.6653
[0d21h01m28s] 2024-09-26T10:13:01 <ERR> [POSTPROC] main: Raw: 579.5727, Value: , Status: Neg. Rate - Read: - Raw: 579.5727 - Pre: 579.6653
```

### Expected Behavior

I would expect the raw value for '6.1' not to be 5 - but 6.

### Screenshots

![image](https://github.com/user-attachments/assets/483bcf3d-b9ab-4cf2-996c-f791f6fff201)

**Comment by p0macs** [comment]
The problem is related to my other issue, see: https://github.com/jomjol/AI-on-the-edge-device/issues/3255
The first analog counter has a mechanical "defect" and does not show the correct position.
On the example picture, the first counter is recognized as "6.1" and the next is "7.3".
The first should be already "6.7" - but because it is closer to "6" - the algorithm probably think it still only approaching to "6" - because the next digit is already "7" - so it decide the first digit to be rather "5".
Unfortunately, I have a lot of wrong readings because of this, about 70% of my readings have "negative rate".

**Comment by DaHype** [comment]
I have all 8-12 hours neg.rate. After Set RAW value,All ok some hours.
I tried already so much with the settings since several month. I don't find the Problem. So my last hope that someone have an Idea what could i try.... Thanks 
![Screenshot_20241021-204327](https://github.com/user-attachments/assets/d8d3d86c-fc85-4230-b364-720eb92f158a)

**Reply by SybexX** [reply]
Apparently there is a problem with the detection (0 <> 9), you could try a different model and see if it gets better.
Also take a look at these parameters: https://jomjol.github.io/AI-on-the-edge-device-docs/Watermeter-specific-analog---digital-transition/ and https://jomjol.github.io/AI-on-the-edge-device-docs/Correction%20Algorithm/#maxratevalue-and-maxratetype
